flink/src/test/java/org/apache/iceberg/flink/actions/TestMigrateAction.java

```diff
@@ -0,0 +1,308 @@
+
+    Schema schema = new Schema(
+        Types.NestedField.required(1, "id", Types.IntegerType.get()),
+        Types.NestedField.optional(2, "data", Types.StringType.get()));
+    GenericAppenderFactory genericAppenderFactory = new GenericAppenderFactory(schema);
+
+    URL url = new URL(location + File.separator + "test." + format.name());
+    File dataFile = new File(url.getPath());
+
+    List<Object[]> expected = Lists.newArrayList();
+    try (FileAppender<Record> fileAppender = genericAppenderFactory.newAppender(Files.localOutput(dataFile), format)) {
+      for (int i = 0; i < 10; i++) {
+        Record record = SimpleDataUtil.createRecord(i, "iceberg");
+        fileAppender.add(record);
+        expected.add(new Object[] {i, "iceberg"});
+      }
+    }
+
+    List<ManifestFile> manifestFiles =
+        Actions.migrateHive2Iceberg(env, flinkHiveCatalog, SOURCE_HIVE_DB_NAME, SOURCE_HIVE_TABLE_NAME,
+            validationCatalog, baseNamespace, DATABASE, TABLE_NAME_UNPARTITIONED).execute();
+    Assert.assertEquals("Should produce the expected manifestFiles count.", 1, manifestFiles.size());
+
+    sql("USE CATALOG %s", catalogName);
+    sql("USE %s", DATABASE);
+    List<Object[]> list = sql("SELECT * FROM %s", TABLE_NAME_UNPARTITIONED);
+    Assert.assertEquals("Should produce the expected records count.", 10, list.size());
+    Assert.assertArrayEquals("Should produce the expected records.", expected.toArray(), list.toArray());
+  }
+
+  @Test
+  public void testMigratePartition() throws IOException, TException, TableNotExistException {
+    getTableEnv().getConfig().setSqlDialect(SqlDialect.HIVE);
+    sql("CREATE TABLE %s (id INT, data STRING) PARTITIONED BY (p STRING) STORED AS %s", SOURCE_HIVE_TABLE_NAME,
+        format.name());
+    getTableEnv().getConfig().setSqlDialect(SqlDialect.DEFAULT);
+    String hiveLocation = flinkHiveCatalog.getHiveTable(new ObjectPath(SOURCE_HIVE_DB_NAME, SOURCE_HIVE_TABLE_NAME))
+        .getSd().getLocation();
+
+    List<Object[]> expected = Lists.newArrayList();
+    String[] partitions = new String[] {"iceberg", "flink"};
+    for (String partitionValue : partitions) {
+      String partitionPath = hiveLocation + "/p=" + partitionValue;
+
+      Partition hivePartition = createHivePartition(format, partitionPath, partitionValue);
+      metastoreClient.add_partition(hivePartition);
+
+      Partition partition =
+          metastoreClient.getPartition(SOURCE_HIVE_DB_NAME, SOURCE_HIVE_TABLE_NAME, "p=" + partitionValue);
+      String location = partition.getSd().getLocation();
+
+      Schema schema = new Schema(
+          Types.NestedField.required(1, "id", Types.IntegerType.get()),
+          Types.NestedField.optional(2, "data", Types.StringType.get()));
+      GenericAppenderFactory genericAppenderFactory = new GenericAppenderFactory(schema);
+
+      URL url = new URL(location + File.separator + "test." + format.name());
+      File dataFile = new File(url.getPath());
+
+      try (
```

Comment:
nit: slightly simpler version.
```
    GenericAppenderHelper dataAppender = new GenericAppenderHelper(table, fileFormat, TEMPORARY_FOLDER);
    final List<Record> batch1 = RandomGenericData.generate(table.schema(), 2, 0L);
    dataAppender.appendToTable(batch1);
```

Reply:
It required an exist iceberg table, now we need to generate data for hive table, not iceberg table

Reply:
That is right. I forgot we are inserting into Hive tables